base/number.jl

```diff
@@ -136,6 +136,52 @@ true
 """
 signbit(x::Real) = x < 0
 
+"""
+    ispositive(x)
+
+Test whether `x > 0`. See also [`isnegative`](@ref).
+
+!!! compat "Julia 1.12"
+    This function requires at least Julia 1.12.
+
+# Examples
+```jldoctest
+julia> ispositive(-4.0)
+false
+
+julia> ispositive(99)
+true
+
+julia> ispositive(0.0)
+false
+```
+"""
+ispositive(x) = x > zero(x) # generic fallback
+ispositive(x::Real) = x > 0
```

Comment:
I probably missed it in the discussions above, but what would be the difference between `x > 0` and `x > zero(x)`?

Reply:
It's not related to the discussions. I just realized
- similar methods only support `Real`.
- https://github.com/JuliaLang/julia/issues/35067 asked for a generic comparison againse `zero(x)` and it's also an example where `x>0` doesn't work.

Reply:
I guess the lowest real type on the `promote` totem pole is `Bool`, so I would suggest `x > false` as the generic case (although for most types, no promotion actually occurs). I agree that `x > zero(x)` doesn't add much and requires a type to explicitly define `zero` (which virtually all will, but it seems like an unnecessary dependency). For types requiring heap allocation (like `BigInt`, although it is covered by a separate specialization), `zero` will necessarily require an allocation whereas `0` or `false` might avoid it when mixed-type comparisons are defined.

In other words, `BigInt` is an example indicating that `x > 0`/`x > false` is likely a more performant fallback than `x > zero(x)`.

Reply:
I was more wondering why having the two different methods to start with, sounds like we should a single good definition, since `Real` isn't much more specific than `Any`.

Reply:
as `BigInt` and `BigFloat` have specialized methods anyway, I guess using `zero(x)` universally is fine.

Reply:
> some types that can behave like reals are not actually `<: Real`. `DualNumbers.Dual <: Number` being such an example.

I don't think the dual numbers are ordered though? So I guess we don't want `ispositive` or `isnegative` defined for dual numbers.